Source/Renderer/RenderState.js

```diff
@@ -272,12 +278,179 @@ define([
                 throw new RuntimeError('renderState.viewport.height must be less than or equal to the maximum viewport height (' + this.maximumViewportHeight.toString() + ').  Check maximumViewportHeight.');
             }
         }
+        //>>includeEnd('debug');
 
 
         this.id = 0;
         this._applyFunctions = [];
     };
 
+    var nextRenderStateId = 0;
+    var renderStateCache = {};
+
+    /**
+     * Validates and then finds or creates an immutable render state, which defines the pipeline
+     * state for a {@link DrawCommand} or {@link ClearCommand}.  All inputs states are optional.  Omitted states
+     * use the defaults shown in the example below.
+     *
+     * @param {Object} options Object with the following properties:
+     * @param [Object} options.context The context.
+     * @param {Object} options.renderState The states defining the render state as shown in the example below.
```

Comment:
Poke around here to learn more about our reference doc: http://usejsdoc.org/index.html